Assets/Scripts/LifeManager.cs

```diff
@@ -50,18 +50,26 @@ public void InflictDamage(int damageAmount)
         {
             Assert.IsTrue(damageAmount >= 0);
             mCurHealth = Mathf.Max(0, mCurHealth - damageAmount);
-            if (mPhotonView.isMine)
-            {
-                EventSystem.OnUpdateHealthBar(mCurHealth);
-            }
+            
             if (mCurHealth == 0)
             {
                 Die();
             }
+            else if (mPhotonView.isMine)
+            {
+                EventSystem.OnUpdateHealthBar(mCurHealth);
+                mPhotonView.RPC("OnDamaged", PhotonTargets.All);

```

Comment:
I'm noticing a noticeable delay on the flashing over the network and I believe its because we are triggering it through this OnDamaged RPC. We should not need to use RPCs for this. Damage and Bullets are simulated locally so we shouldn't need to send out events. Once the InflictDamage method call happens we know a hit has been registered and we shouldn't need any rpcs